### DISCUSSION TITLE
BLE Hosting : Getting the perhiperalid

### DISCUSSION THREAD

**Discussion by FrankVDL2471** [shinyorg/shiny#1354] [Q&A] [answered]
How do I get the Perhiperal uid for the current device from the IBleHostingManager ?

I want to stuff the uid in a QRCode so when scanned by the client app, it can use the GetKnownPeripheral() method to make a connection.
so there is no need for scanning.

**Comment by aritchie** [comment, marked as answer]
You can't do it this way because it isn't something either platform offers at the central side.  The UUID on the client side also isn't a cross platform solution.  You should never store it or use it beyond the local device.  What you can do is control the advertised service UUID on the central and "stuff that in a QR code".  Then have your client look for the first peripheral broadcasting that service UUID.  This would be an xplat solution and is easy to achieve with Shiny
